.github/workflows/central_publish.yml

```diff
@@ -0,0 +1,76 @@
+name: Publish to the Ballerina central
+
+on:
+  workflow_dispatch:
+    inputs:
+      environment:
+        type: choice
+        description: Select environment
+        required: true
+        options:
+          - CENTRAL
+          - DEV CENTRAL
+          - STAGE CENTRAL
+
+jobs:
+  central_publish:
+    name: Run central publish workflow
+    runs-on: ubuntu-latest
+
+    if: github.repository_owner == 'ballerina-platform'
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '17.0.7'
+
+      - name: Set Up Ballerina
+        uses: ballerina-platform/setup-ballerina@v1.1.0
+        with:
+          version: latest
+
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+
+      - name: Build without tests
+        env:
+          packageUser: ${{ github.actor }}
+          packagePAT: ${{ secrets.GITHUB_TOKEN }}
+        run: ./gradlew build -x check -x test
+
+      - name: Ballerina Central Push
+        if: ${{ inputs.environment == 'CENTRAL' }}
+        env:
+          BALLERINA_DEV_CENTRAL: false
+          BALLERINA_STAGE_CENTRAL: false
+          BALLERINA_CENTRAL_ACCESS_TOKEN: ${{ secrets.BALLERINA_CENTRAL_ACCESS_TOKEN }}
+          packageUser: ${{ secrets.BALLERINA_BOT_USERNAME }}
+          packagePAT: ${{ secrets.BALLERINA_BOT_TOKEN }}
+          GITHUB_TOKEN: ${{ secrets.BALLERINA_BOT_TOKEN }}
+        run: ./gradlew release -PpublishToCentral=true
```

Comment:
I am assuming you are using the Ballerina Gradle plugin here. In that case, you don't need the Gradle release task here. You can invoke the `build` task with `-P publishToCentral=true`.

We have two separate workflows for release a package and to publish it to the Ballerina central. Following are the reasons for that:

1. The `publish-release` workflow is the main workflow we use to release a package. This workflow should handle the following:
    i. Create a git tag on GitHub
    ii. Publish the relevant Maven artifacts to GitHub packages
   iii. Publish the Ballerina package to the Ballerina Central
    Additionally, it should send a post-release PR to the default branch after the release to sync the default branch with the release branch. So the `publish-release` workflow should be the place to handle Gradle `release` task.
2. The `central-publish` workflow is mainly used to publish the package to the Ballerina Dev Central and the Ballerina Stage Central. (Additionally is is used to publish the package to the Ballerina Central if the publishing to the Ballerina Central is failed in the `publish-release` workflow due to some intermittent issue).

If we use the Gradle `release` task here, it will remove the `SNAPSHOT` version from the `gradle.properties` file and commit it to the relevant branch, which shouldn't happen in this workflow. So In this case, you only need to use the `-P publishToCentral=true`.

This is true for all three publishing steps in this workflow.

```suggestion
        run: ./gradlew clean build -PpublishToCentral=true
```

Reply:
Since this is a standalone Ballerina package, the Ballerina Gradle plugin will not be used. Instead, the CLI-based configurations to push the tool will be introduced.